kratos/input_output/unv_output.cpp

```diff
@@ -0,0 +1,226 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                     Kratos default license: kratos/license.txt
+//
+//  Main authors:    Carlos Roig
+//
+//
+
+/* System includes */
+#include <iostream>
+#include <fstream>
+
+/* External includes */
+
+/* Project includes */
+#include "includes/model_part.h"
+#include "includes/exception.h"
+
+#include "input_output/unv_output.h"
+
+namespace Kratos {
+
+UnvOutput::UnvOutput(Kratos::ModelPart &modelPart, const std::string &outFileWithoutExtension)
+  : mrOutputModelPart(modelPart),
+    mOutputFileName(outFileWithoutExtension + ".unv") {
+}
+
+
+void UnvOutput::WriteMesh() {
+    WriteNodes();
+    WriteElements();
+}
+
+template <typename Enumeration>
+auto as_integer(Enumeration const value)
+    -> typename std::underlying_type<Enumeration>::type
+{
+    return static_cast<typename std::underlying_type<Enumeration>::type>(value);
+}
+
+void UnvOutput::InitializeOutputFile() {
+    std::ofstream outputFile;
+    outputFile.open(mOutputFileName, std::ios::out | std::ios::trunc);
+    outputFile.close();
+}
+
+void UnvOutput::WriteNodes() {
+    std::ofstream outputFile;
+    outputFile.open(mOutputFileName, std::ios::out | std::ios::app);
+
+    outputFile << std::scientific;
+    outputFile << std::setprecision(15);
+
+    const int dataSetNumberForNodes = 2411;
```

Comment:
This value?

Reply:
This is the id of the dataset. 2411 -> nodes, 2412 -> Elements, 2414 -> Results,

I will move it a enum as well.